earth_enterprise/rpms/shared/install-utils.sh.template

```diff
@@ -32,22 +32,43 @@ ASSET_ROOT="/gevol/assets"
 SOURCE_VOLUME="/gevol/src"
 
 # Commands:
+CHKCONFIG="/sbin/chkconfig"
+INITSCRIPT_UPDATE="/usr/sbin/update-rc.d"
 NEWLINECLEANER="sed -e s:\\n::g"
 
 
 <%
 out << '''
 load_systemrc_config()
 {
+    local VALUE
+
     if [ -f "$SYSTEMRC" ]; then
         # Read configuration:
-        ASSET_ROOT=$(xmllint --xpath '//Systemrc/assetroot/text()' "$SYSTEMRC")
-        GEFUSIONUSER=$(xmllint --xpath '//Systemrc/fusionUsername/text()' "$SYSTEMRC")
-        GEGROUP=$(xmllint --xpath '//Systemrc/userGroupname/text()' "$SYSTEMRC")
+        VALUE=$(xml_file_get_xpath "$SYSTEMRC" "//Systemrc/assetroot/text()")
+        [ -z "$VALUE" ] || ASSET_ROOT="$VALUE"
+
+        VALUE=$(xml_file_get_node_text "$SYSTEMRC" "//Systemrc/fusionUsername/text()")
+        [ -z "$VALUE" ] || GEFUSIONUSER="$VALUE"
+
+        VALUE=$(xml_file_get_node_text "$SYSTEMRC" "//Systemrc/userGroupname/text()")
+        [ -z "$VALUE" ] || GEGROUP="$VALUE"
     fi
 }
+
+xml_file_get_xpath()
+{
+    local FILE="$1"
+    local XPATH="$2"
+
+    # Warning: `xmllint --noent` doesn't recognize named entities like
+    # "&lt;" and "&gt;":
+    echo "cat $XPATH" | xmllint --noent --nocdata --shell "$FILE" |
+    # Skip the first and the last line:
```

Comment:
Minor: this comment is now wrong.

Reply:
Which comment is wrong?

Reply:
"Skip the first and the last line" since we're now skipping the last 2 lines.  It's a small thing.

Reply:
The comment is correct. The code wasn't running correctly before.